####The VCF files, including 30 H. hainanensis samples and 32 H. reticulata samples were called using the longest two monoploid genomes as the reference genome.
####The VCF files were split into different chromosomes. SMC++ merges the results of different chromosomes in the subsequent analysis. 
####The analysis was run for species H. hainanensis and H. reticulata separately. 
bcftools index ./Hret.Chr1.vcf.gz
bcftools index ./Hret.Chr2.vcf.gz
bcftools index ./Hret.Chr3.vcf.gz
bcftools index ./Hret.Chr4.vcf.gz
bcftools index ./Hret.Chr5.vcf.gz
bcftools index ./Hret.Chr6.vcf.gz
bcftools index ./Hret.Chr7.vcf.gz
bcftools index ./Hret.Chr8.vcf.gz
bcftools index ./Hret.Chr9.vcf.gz
bcftools index ./Hret.Chr10.vcf.gz
bcftools index ./Hret.Chr11.vcf.gz
bcftools index ./Hret.Chr12.vcf.gz
bcftools index ./Hret.Chr13.vcf.gz
bcftools index ./Hret.Chr14.vcf.gz

bcftools index ./Hhai.Chr1.vcf.gz
bcftools index ./Hhai.Chr2.vcf.gz
bcftools index ./Hhai.Chr3.vcf.gz
bcftools index ./Hhai.Chr4.vcf.gz
bcftools index ./Hhai.Chr5.vcf.gz
bcftools index ./Hhai.Chr6.vcf.gz
bcftools index ./Hhai.Chr7.vcf.gz
bcftools index ./Hhai.Chr8.vcf.gz
bcftools index ./Hhai.Chr9.vcf.gz
bcftools index ./Hhai.Chr10.vcf.gz
bcftools index ./Hhai.Chr11.vcf.gz
bcftools index ./Hhai.Chr12.vcf.gz
bcftools index ./Hhai.Chr13.vcf.gz
bcftools index ./Hhai.Chr14.vcf.gz
###step2 This subcommand converts vcd data into the format used by SMC++, transforming the vcd format into SMC format, where "sample" is the sample name.
smc++ vcf2smc ./Hret.Chr1.vcf.gz out/Hret.Chr1.smc.gz chr1 Hret:sample1,sample2,... 
smc++ vcf2smc ./Hret.Chr2.vcf.gz out/Hret.Chr2.smc.gz chr2 Hret:sample1,sample2,...
smc++ vcf2smc ./Hret.Chr3.vcf.gz out/Hret.Chr3.smc.gz chr3 Hret:sample1,sample2,...
smc++ vcf2smc ./Hret.Chr4.vcf.gz out/Hret.Chr4.smc.gz chr4 Hret:sample1,sample2,...
smc++ vcf2smc ./Hret.Chr5.vcf.gz out/Hret.Chr5.smc.gz chr5 Hret:sample1,sample2,...
smc++ vcf2smc ./Hret.Chr6.vcf.gz out/Hret.Chr6.smc.gz chr6 Hret:sample1,sample2,...
smc++ vcf2smc ./Hret.Chr7.vcf.gz out/Hret.Chr7.smc.gz chr7 Hret:sample1,sample2,...
smc++ vcf2smc ./Hret.Chr8.vcf.gz out/Hret.Chr8.smc.gz chr8 Hret:sample1,sample2,...
smc++ vcf2smc ./Hret.Chr9.vcf.gz out/Hret.Chr9.smc.gz chr9 Hret:sample1,sample2,...
smc++ vcf2smc ./Hret.Chr10.vcf.gz out/Hret.Chr10.smc.gz chr10 Hret:sample1,sample2,...
smc++ vcf2smc ./Hret.Chr11.vcf.gz out/Hret.Chr11.smc.gz chr11 Hret:sample1,sample2,...
smc++ vcf2smc ./Hret.Chr12.vcf.gz out/Hret.Chr12.smc.gz chr12 Hret:sample1,sample2,...
smc++ vcf2smc ./Hret.Chr13.vcf.gz out/Hret.Chr13.smc.gz chr13 Hret:sample1,sample2,...
smc++ vcf2smc ./Hret.Chr14.vcf.gz out/Hret.Chr14.smc.gz chr14 Hret:sample1,sample2,...
smc++ vcf2smc ./Hhai.Chr1.vcf.gz out/Hhai.Chr1.smc.gz chr1 Hhai:sample1,sample2,...
smc++ vcf2smc ./Hhai.Chr2.vcf.gz out/Hhai.Chr2.smc.gz chr2 Hhai:sample1,sample2,...
smc++ vcf2smc ./Hhai.Chr3.vcf.gz out/Hhai.Chr3.smc.gz chr3 Hhai:sample1,sample2,...
smc++ vcf2smc ./Hhai.Chr4.vcf.gz out/Hhai.Chr4.smc.gz chr4 Hhai:sample1,sample2,...
smc++ vcf2smc ./Hhai.Chr5.vcf.gz out/Hhai.Chr5.smc.gz chr5 Hhai:sample1,sample2,...
smc++ vcf2smc ./Hhai.Chr6.vcf.gz out/Hhai.Chr6.smc.gz chr6 Hhai:sample1,sample2,...
smc++ vcf2smc ./Hhai.Chr7.vcf.gz out/Hhai.Chr7.smc.gz chr7 Hhai:sample1,sample2,...
smc++ vcf2smc ./Hhai.Chr8.vcf.gz out/Hhai.Chr8.smc.gz chr8 Hhai:sample1,sample2,...
smc++ vcf2smc ./Hhai.Chr9.vcf.gz out/Hhai.Chr9.smc.gz chr9 Hhai:sample1,sample2,...
smc++ vcf2smc ./Hhai.Chr10.vcf.gz out/Hhai.Chr10.smc.gz chr10 Hhai:sample1,sample2,...
smc++ vcf2smc ./Hhai.Chr11.vcf.gz out/Hhai.Chr11.smc.gz chr11 Hhai:sample1,sample2,...
smc++ vcf2smc ./Hhai.Chr12.vcf.gz out/Hhai.Chr12.smc.gz chr12 Hhai:sample1,sample2,...
smc++ vcf2smc ./Hhai.Chr13.vcf.gz out/Hhai.Chr13.smc.gz chr13 Hhai:sample1,sample2,...
smc++ vcf2smc ./Hhai.Chr14.vcf.gz out/Hhai.Chr14.smc.gz chr14 Hhai:sample1,sample2,...
####step3.step4 Estimate population size history and plots fitted size histories. Draw population plots for different mutation rates.
smc++ cv --timepoints 10 10000000 -o Hhai/analysis_15/   1.05707e-08   out/Hhai.Chr*.smc.gz
smc++ plot -g 15  Hhai.smc.pdf Hhai/analysis_15/model.final.json
smc++ cv --timepoints 10 10000000 -o Hhai/analysis_20/   1.40943E-08   out/Hhai.Chr*.smc.gz
smc++ plot -g 20  Hhai.smc.pdf Hhai/analysis_20/model.final.json
smc++ cv --timepoints 10 10000000 -o Hhai/analysis_30/   2.11414E-08   out/Hhai.Chr*.smc.gz
smc++ plot -g 30  Hhai.smc.pdf Hhai/analysis_30/model.final.json

smc++ cv --timepoints 10 10000000 -o Hret/analysis_15/   1.05707e-08   out/Hret.Chr*.smc.gz
smc++ plot -g 15  Hret.smc.pdf  Hret/analysis_15/model.final.json
smc++ cv --timepoints 10 10000000 -o Hret/analysis_20/   1.40943E-08   out/Hret.Chr*.smc.gz
smc++ plot -g 20  Hret.smc.pdf  Hret/analysis_20/model.final.json
smc++ cv --timepoints 10 10000000 -o Hret/analysis_30/   2.11414E-08   out/Hret.Chr*.smc.gz
smc++ plot -g 30  Hret.smc.pdf  Hret/analysis_30/model.final.json
